packages/arb-bridge-eth/contracts/arch/Value.sol

```diff
@@ -268,11 +244,11 @@ library Value {
         return newTuple(values);
     }
 
-    function newHashOnly(bytes32 _val) internal pure returns (Data memory) {
-        return Data(uint256(_val), CodePoint(0, 0, false, 0), new Data[](0), HASH_ONLY_TYPECODE);
+    function newHashOnly(bytes32 _val, uint256 size) internal pure returns (Data memory) {
+        return Data(uint256(_val), CodePoint(0, 0, false, 0), new Data[](0), HASH_ONLY_TYPECODE, size);
     }
 
-    function deserializeHashOnly(
+    function deserializeHashValue(
```

Comment:
This allows you to submit an arbitrary size and hash. It seems like there are contexts where this data is not validated and the caller could submit a size that's not correct for the given hash.